pipelines/tekton/pipeplines_provider.go

```diff
@@ -112,6 +118,16 @@ func (pp *PipelinesProvider) Run(ctx context.Context, f fn.Function) error {
 		}
 	}
 
+	if f.Build.Git.URL == "" {
```

Comment:
@lkingland had some idea on this^

Reply:
In this PR am I right to understand that `func deploy --remote` with no git URL specified is the proper way to force upload? That's a nice start, but agree that it would be better if the pipeline provider here could use explicit configuration to determine whether it's push or pull.

Reply:
@lance @zroubalik @lkingland could we for now proceed with simplified version where direct upload is used only if git-uri is not set?

Reply:
I think we can, we just need to track that effort and implement it before next release imho, so we don't change the behavior many times.

Reply:
Yes, that is correct.  Please see related issue https://github.com/knative/func/issues/1075  which explains the choice of flag permutations in detail.
In short:  `func` presumes you are operating on the code in the current working directory, or at `--path`.  So the use of --git-url is actually the exceptional case, and this PR is implementing the default case, which is "deploy the current function using a remote build", which is simply:
```
func deploy --remote
```
Your pr was already correct in that it simply removes the error condition which says "sorry, this isn't implemented yet, , please use --git-url"

Reply:
> Would be nice, if I can have the git url specified in the `func.yaml`, but chose to use the direct upload for the particular `func deploy` call. ie. I as a developer want to instruct func to use direct code upload and have git url in `func.yaml` specified at the same time. 

Yes, this is a known limitation which has come up a few times:  Developer want's flag values to persist to the func.yaml for the next call by default, but sometimes choose to do something difference, but "not save".   I think a simple solution to this will be to implement a `--no-save` flag which, when provided, skips `f.Write()` entirely  (or perhaps writes only a subset of the flag values).  Unfortunately #1283 is a prerequisite.

Reply:
@lkingland so this works as expected and we could merge this? And followup PR will take care of persisting the git flag?

Reply:
>The PR I see as being correct in that it removes the error condition which essentially says "sorry, this isn't implemented yet, , please provide --git-url".

Yes, that was my initial thought.

Reply:
It's my opinion that yes, this is GTG as -is, and both @lance and @zroubalik 's comments are addressed at length in the associated issue, which I think would be best addressed in a separate PR that (I hope) adds a `--save` flag.

Reply:
wdyt @lance @zroubalik ?

Reply:
yeah let's proceed with this one and tackle the ux in some following PR